packages/cdktf/lib/terraform-hcl-module.ts

```diff
@@ -0,0 +1,81 @@
+import { Construct } from "constructs";
+import { TerraformModuleOptions, TerraformModule } from "./terraform-module";
+import { TerraformProvider } from "./terraform-provider";
+
+export interface TerraformHclModuleOptions extends TerraformModuleOptions {
+    readonly providers?: (TerraformProvider | TerraformModuleProvider)[];
+    readonly variables?: {[key: string]: any};
```

Comment:
It's a bit unfortunate to name this, but it won't translate well to java/c# if left as an indexer.